### PR TITLE
Add nested CVM attestation testcase

### PR DESCRIPTION
**Pull request by smit-gardhariya**
This PR will create new testcase which will verify attestation report for nested CVM guest.

### CODE REVIEW

File: microsoft/testsuites/cvm_attestation/cvm_attestation.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        host_data = variables.get("host_data", "")
+        if not host_data:
+            raise LisaException("host_data is empty")
```
Comment (squirrelsc) [inline microsoft/testsuites/cvm_attestation/cvm_attestation.py]: If it's empty, the test case should be skipped, instead of failed.

Comment (smit-gardhariya) [reply microsoft/testsuites/cvm_attestation/cvm_attestation.py]: @squirrelsc Done with latest push

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

File: microsoft/testsuites/cvm_attestation/cvm_attestation_tool.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    def _install(self) -> bool:
+        echo = self.node.tools[Echo]
+        git = self.node.tools[Git]
+
+        root_path = PurePath(
+            echo.run(
+                "$HOME",
+                shell=True,
+                expected_exit_code=0,
+                expected_exit_code_failure_message="failed to get $HOMW via echo",
```
Comment (LiliDeng) [inline microsoft/testsuites/cvm_attestation/cvm_attestation_tool.py]: typo $HOMW

File: microsoft/testsuites/cvm_attestation/cvm_attestation_tool.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        echo = self.node.tools[Echo]
+
+        tool_path = PurePath(
+            echo.run(
+                "$HOME",
+                shell=True,
+                expected_exit_code=0,
+                expected_exit_code_failure_message="failed to get $HOMW via echo",
```
Comment (LiliDeng) [inline microsoft/testsuites/cvm_attestation/cvm_attestation_tool.py]: typo $HOMW

File: microsoft/testsuites/cvm_attestation/cvm_attestation.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+)
+class CVMAttestationTestSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node = kwargs["node"]
+        sev_guest_exists = node.tools[Ls].path_exists(
+            path="/dev/sev-guest",
+            sudo=True,
+        )
+        if not sev_guest_exists:
+            raise LisaException("/dev/sev-guest: Device Not Found")
```
Comment (LiliDeng) [inline microsoft/testsuites/cvm_attestation/cvm_attestation.py]: this case will always fail in azure platform

File: microsoft/testsuites/cvm_attestation/cvm_attestation_tool.py (hunk 2 above)
Comment (smit-gardhariya) [reply microsoft/testsuites/cvm_attestation/cvm_attestation_tool.py]: @LiliDeng Corrected in latest push

File: microsoft/testsuites/cvm_attestation/cvm_attestation_tool.py (hunk 3 above)
Comment (smit-gardhariya) [reply microsoft/testsuites/cvm_attestation/cvm_attestation_tool.py]: @LiliDeng Corrected in latest push

File: microsoft/testsuites/cvm_attestation/cvm_attestation.py (hunk 4 above)
Comment (smit-gardhariya) [reply microsoft/testsuites/cvm_attestation/cvm_attestation.py]: @LiliDeng 

Correct, This testcase is for nested CVM only. I wanted to fail the testcase if device is not there so we can verify the nested CVM CLH scenarios too.

I have changed exception to SkippedException. Now it wont affect the azure platform.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**